fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/Deskolemizer.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.kernel.impl.rdf;
+
+import static com.google.common.cache.CacheBuilder.newBuilder;
+import static com.google.common.cache.CacheLoader.from;
+import static com.hp.hpl.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.fcrepo.kernel.FedoraJcrTypes.FEDORA_SKOLEMNODE;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import javax.jcr.AccessDeniedException;
+
+import org.fcrepo.kernel.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.identifiers.IdentifierConverter;
+import org.fcrepo.kernel.models.FedoraResource;
+
+import org.slf4j.Logger;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.cache.LoadingCache;
+import com.hp.hpl.jena.graph.Triple;
+import com.hp.hpl.jena.rdf.model.Model;
+import com.hp.hpl.jena.rdf.model.RDFNode;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.Statement;
+
+/**
+ * Deskolemization is abstractly a function from RDF nodes to RDF nodes, but here we implement it, purely for
+ * convenience of operation, as a function from triples to triples. This object should be used to translate only one
+ * document's scope of RDF.
+ *
+ * @author ajs6f
+ */
+public class Deskolemizer implements Function<Triple, Triple> {
```

Comment:
Yes, this is an oversight. As far as the name goes, I don't know how much difference it makes. The only reason that there is a `Deskolemizer` interface is because the workings of this process tangle up into the kernel API, not because we have any reasonable expectation of more than one implementation of deskolemization. There is no clean way to introduce blank nodes, and this code is crap, but it's as good as I think we can expect to do. If you think it clearer, I can certainly change the name here to `DeskolemizerImpl`.